src/dag/dag_block_manager.cpp

```diff
@@ -0,0 +1,268 @@
+    for (auto &it : unverified_qu_) {
+      res.first += it.second.size();
+    }
+  }
+
+  {
+    uLock lock(shared_mutex_for_verified_qu_);
+    res.second = 0;
+    for (auto &it : verified_qu_) {
+      res.second += it.second.size();
+    }
+  }
+  return res;
+}
+
+DagBlock DagBlockManager::popVerifiedBlock() {
+  uLock lock(shared_mutex_for_verified_qu_);
+  while (verified_qu_.empty() && !stopped_) {
+    cond_for_verified_qu_.wait(lock);
+  }
+  if (stopped_) return DagBlock();
+
+  auto blk = verified_qu_.begin()->second.front();
+  verified_qu_.begin()->second.pop_front();
+  if (verified_qu_.begin()->second.empty()) verified_qu_.erase(verified_qu_.begin());
+  return blk;
+}
+
+void DagBlockManager::pushVerifiedBlock(DagBlock const &blk) {
+  uLock lock(shared_mutex_for_verified_qu_);
+  verified_qu_[blk.getLevel()].emplace_back(blk);
+}
+
+void DagBlockManager::verifyBlock() {
+  while (!stopped_) {
+    std::pair<DagBlock, std::vector<Transaction>> blk;
+    {
+      uLock lock(shared_mutex_for_unverified_qu_);
+      while (unverified_qu_.empty() && !stopped_) {
+        cond_for_unverified_qu_.wait(lock);
+      }
+      if (stopped_) {
+        return;
+      }
+      blk = unverified_qu_.begin()->second.front();
+      unverified_qu_.begin()->second.pop_front();
+      if (unverified_qu_.begin()->second.empty()) {
+        unverified_qu_.erase(unverified_qu_.begin());
+      }
+    }
+    auto status = blk_status_.get(blk.first.getHash());
+    // Verifying transaction ...
+    LOG(log_time_) << "Verifying Trx block  " << blk.first.getHash() << " at: " << getCurrentTimeMilliSeconds();
+    // only need to verify if this is a broadcasted block (proposed block are generated by verified trx)
+    if (!status.second || status.first == BlockStatus::broadcasted) {
+      // Verify transactions
+      if (!trx_mgr_->verifyBlockTransactions(blk.first, blk.second)) {
+        LOG(log_er_) << "Ignore block " << blk.first.getHash() << " since it has invalid or missing transactions";
+        blk_status_.update(blk.first.getHash(), BlockStatus::invalid);
+        continue;
+      }
+      // Verify VDF solution
+      vdf_sortition::VdfSortition vdf = blk.first.getVdf();
+      if (!vdf.verifyVdf(vdf_config_, getRlpBytes(blk.first.getLevel()), blk.first.getPivot().asBytes())) {
+        LOG(log_er_) << "DAG block " << blk.first.getHash() << " failed on VDF verification with pivot hash "
+                     << blk.first.getPivot();
+        blk_status_.update(blk.first.getHash(), BlockStatus::invalid);
+        continue;
+      }
+      // Verify DPOS
+      auto period = getPeriod(blk.first.getLevel());
+      auto dag_block_sender = blk.first.getSender();
+      if (!final_chain_->dpos_is_eligible(period, dag_block_sender)) {
+        auto executed_period = pbft_chain_->getPbftExecutedChainSize();
+        auto dpos_period = executed_period;
+        if (dpos_config_) {
+          dpos_period += dpos_config_->deposit_delay;
+        }
```

Comment:
would be acutally nice to use some new lines to separate some blocks of code, it is easier to orientate in code and read it...

Reply:
I agree with you, but I don't think should add new lines here